thirdeye/thirdeye-hadoop/src/main/java/com/linkedin/thirdeye/hadoop/push/SegmentPushControllerAPIs.java

```diff
@@ -174,18 +174,17 @@ private String deleteSegment(String tablename, String segmentName) throws IOExce
         + DROP_PARAMETERS);
     HttpResponse res = controllerClient.execute(controllerHttpHost, req);
     try {
-      if (res.getStatusLine().getStatusCode() != 200) {
-        throw new IllegalStateException(res.getStatusLine().toString());
+      if (res == null || res.getStatusLine() == null || res.getStatusLine().getStatusCode() != 200) {
```

Comment:
convert this into a retry loop and retry 3 times and have a sleep between re tries.